caput/pipeline/runner/_core.py

```diff
@@ -1,19 +1,32 @@
-"""caput.pipeline.runner."""
+"""Core pipeline runner functions.
+
+These don't provide much functionality on their own, but are used
+in the command-line interface.
+"""
+
+from __future__ import annotations
 
 import logging
 import sys
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    import os
+    from collections.abc import Generator
+    from typing import Any, Literal
+
 
 logger = logging.getLogger(__name__)
 # Set the logging level and format
 logging.basicConfig(level=logging.INFO, format="%(message)s")
 
 
-def lint_config(configfile):
+def lint_config(configfile: os.PathLike | tuple[os.PathLike, ...]) -> None:
     """Lint a pipeline config file without running it.
 
     Parameters
     ----------
-    configfile : str or tuple of str
+    configfile : os.PathLike
```

Comment:
```suggestion
    configfile : os.PathLike | tuple[os.PathLike, ...]
```
For consistency.

(Also the render is replacing the `...`, with its type (`Ellipsis`), which doesn't seem correct, but probably not worth figuring out.)

Reply:
Oh, that's annoying. I should figure that out...